mono/metadata/cominterop.c

```diff
@@ -3411,20 +3411,19 @@ cominterop_release_all_rcws (void)
 }
 
 gpointer
-mono_string_to_bstr (MonoString *string_obj)
+mono_string_to_bstr (gpointer ptr, int slen)
```

Comment:
Instead of changing this function, which is used by the generated wrappers in marshal.c

Introduce a mono_ptr_to_bstr local function here that is them called by mono_string_to_bstr.